src/components/Dialogs/AddLinkTdDialog.jsx

```diff
@@ -0,0 +1,294 @@
+/********************************************************************************
+ * Copyright (c) 2018 - 2021 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the W3C Software Notice and
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR W3C-20150513
+ ********************************************************************************/
+ import React, { forwardRef, useContext, useEffect, useImperativeHandle,useCallback } from 'react';
+ import ReactDOM from "react-dom";
+ import ediTDorContext from "../../context/ediTDorContext";
+ import {hasLinks,checkIfLinkIsInItem } from "../../util.js";
+ import { DialogTemplate } from "./DialogTemplate";
+
+ export const AddLinkTdDialog = forwardRef((props, ref) => {
+     const context = useContext(ediTDorContext);
+     const [display, setDisplay] = React.useState(() => { return false });
+     const [linkingMethod, setlinkingMethod] = React.useState(() => { return "url" });
+     const [currentLinkedTd, setCurrentLinkedTd] = React.useState(() => { return {} });
+
+
+     const interaction = props.interaction ?? {};
+     const offlineTD = JSON.parse(context.offlineTD)
+
+     useEffect(() => {
+     }, [display, context]);
+
+     useImperativeHandle(ref, () => {
+         return {
+             openModal: () => open(),
+             close: () => close()
+         }
+     });
+
+     const open = () => {
+         setDisplay(true)
+     };
+
+     const close = () => {
+         setDisplay(false);
+     };
+
+
+     const checkIfLinkExists = (link) => {
+         if (hasLinks(interaction)) {
+            return checkIfLinkIsInItem(link, interaction)
+         }
+         return false
+     }
+
+     const addLinksToInteraction = (link) => {
+                let td = {}
+                 try {
+                     td = JSON.parse(context.offlineTD);
+                 } catch (_) { }
+                 if (!hasLinks(td)) {
+                     td.links = [];
+                 }
+                 td.links.push(link);
+                 context.updateOfflineTD(JSON.stringify(td, null, 2))
+     }
+
+     const getFileHandle = () => {
```

Comment:
Isn`t that method already in AppHeader.jsx? 
Maybe a good candidate for moving to util.js in order to have duplicated code for opening files?